### DISCUSSION TITLE
Provide a button for admins to export the list of members as a CSV file.

### DISCUSSION THREAD

**Discussion by Eugene-Song** [open-webui/open-webui#3640] [General]
Just added a button in admin panel and database section to export the list of members as a CSV file.
<img width="545" alt="截屏2024-07-05 上午12 54 31" src="https://github.com/open-webui/open-webui/assets/63761776/926b5615-269b-4bd1-93d4-3d414355d0d1">

**Comment by Eugene-Song** [comment]
<img width="797" alt="Screenshot 2024-07-05 at 1 02 21 AM" src="https://github.com/open-webui/open-webui/assets/63761776/053a0f32-4c18-4b15-b85d-139c8d96ddcc">
And a screenshot for downloaded file.

**Comment by pierrelouisbescond** [comment]
@Eugene-Song I was looking for this feature. It was not approved/merged with the main branch in the end?

**Reply by hdnh2006** [reply]
It looks like it wasn't unfortunately, the only option is to use the api

**Reply by pierrelouisbescond** [reply]
For those interested, here is the code to retrieve the users' list from the API:

```
import requests

url = 'https://xxx.xxx.xxx/api/v1/users/all'

headers = {
    'Authorization': 'Bearer YOUR_API_KEY',
    'accept': 'application/json'
}

response = requests.get(url, headers=headers)

if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print(f"Request failed with status code: {response.status_code}")
    print(f"Response content: {response.text}")
```
